lib/net.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+const net = require('net');
+
+const sock = require('./socket');
+const transportCommon = require('./transport-common');
+const jstpServer = require('./server');
+const common = require('./common');
+
+class Server extends net.Server {
+  // Constructs JSTP Server bound to node net.Server.
+  //   options - an object that contains applications array or
+  //             index and can optionally contain authPolicy and
+  //             heartbeatInterval (see jstp.Server).
+  //             Also it will be passed directly to node net.Server.
+  //   listener - request listener that will be passed directly to
+  //              net.Server
+  //
+  // See node net.Server for options docs.
+  //
+  constructor(options, listener) {
```

Comment:
This shouldn't be a listener of a raw TCP connection, that's practically useless and is a leaky abstraction. The listener function that is passed to `super()` must be an internal function, while the optional `listener` argument of this constructor should be attached as a listener of the `'connect'` event, which shall be introduced.